django/test/signals.py

```diff
@@ -178,3 +178,10 @@ def static_finders_changed(**kwargs):
     }:
         from django.contrib.staticfiles.finders import get_finder
         get_finder.cache_clear()
+
+
+@receiver(setting_changed)
+def auth_user_model_changed(**kwargs):
+    if kwargs['setting'] == 'AUTH_USER_MODEL':
+        from django.apps import apps
+        apps.get_swappable_setting.cache_clear()
```

Comment:
This signal receiver is only meant for using `override_settings()` in the testing suite. If you ever change it outside the testing suite, you indeed have a lot more to deal with, but the aim of this code is not to cover that case. I will take another look at `get_swappable_settings` and try to live up to the standards in django.apps.